### PR TITLE
Support checking HTTP Response Header

### PR DESCRIPTION
**Pull request by goccy**
Enable checking HTTP response header .

`header` accepts `map[string]string` or `map[string][]string` type as key/value pair .

```yaml
  expect:
    header:
      Content-Length: 30
```

or 

```yaml
  expect:
    header:
      Content-Length:
        - 0
```

### CODE REVIEW

File: protocol/http/expect.go [hunk 1]
```diff
@@ -24,20 +26,37 @@ func (e *Expect) Build(ctx *context.Context) (assert.Assertion, error) {
 	assertion := assert.Build(expectBody)
 
 	return assert.AssertionFunc(func(v interface{}) error {
-		res, ok := v.(*result)
+		res, ok := v.(response)
 		if !ok {
-			return errors.Errorf("expected *result but got %T", v)
+			return errors.Errorf("expected response but got %T", v)
+		}
+		if err := e.assertHeader(res.Header); err != nil {
```
Comment (zoncoen) [inline protocol/http/expect.go:33]: Could you move `assertHeader()` after `assertCode()`?

File: protocol/http/request.go [hunk 2]
```diff
@@ -84,25 +85,25 @@ func (r *Request) Invoke(ctx *context.Context) (*context.Context, interface{}, e
...
+		status: resp.Status,
+	}
 	if len(b) > 0 {
 		unmarshaler := unmarshaler.Get(resp.Header.Get("Content-Type"))
+		var respBody interface{}
 		if err := unmarshaler.Unmarshal(b, &respBody); err != nil {
 			return ctx, nil, errors.Errorf("failed to unmarshal response body as %s: %s: %s", unmarshaler.MediaType(), string(b), err)
 		}
-
-		ctx = ctx.WithResponse(respBody)
```
Comment (zoncoen) [inline protocol/http/request.go:94]: Could you keep this line? `ctx = ctx.WithResponse(rvalue)` is a breaking change.
It changes the way to refer response body value in the template. (e.g. `{{response.messageId}}` => `{{response.body.messageId}}`

File: protocol/http/expect.go (hunk 1 above)
Comment (goccy) [reply protocol/http/expect.go]: I think I've fixed it !

File: protocol/http/request.go (hunk 2 above)
Comment (goccy) [reply protocol/http/request.go:94]: I think I've fixed it , Thanks !

**Pull request merged**